### ISSUE TITLE
FALSE POSITIVE: FindOpenStream does not account for closure of a `PrintStream`'s underlying `FileOutputStream`

### ISSUE THREAD

**Issue by lynae99** [spotbugs/spotbugs#4304]
Hi, I found a false positive in SpotBugs 4.10.3 because FindOpenStream does not account for closure of a `PrintStream`'s underlying `FileOutputStream`.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs OS_OPEN_STREAM`

**Minimal reproducer**

```java
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.PrintStream;

class TestFlushedPrintStreamUnderlyingClose {
    void test() throws IOException {
        FileOutputStream stream = new FileOutputStream("file.txt");
        PrintStream writer = new PrintStream(stream);

        writer.println("value");
        writer.flush();

        stream.close();
    }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-os-open-stream-fp-underlying-close.java
spotbugs -textui -effort:max -low -visitors FindOpenStream classes
```

**Current behavior**

SpotBugs reports `OS_OPEN_STREAM` at line 8, where the `PrintStream` is constructed.

**Expected behavior**

SpotBugs should not report `OS_OPEN_STREAM` because the wrapper is flushed and its underlying `FileOutputStream` is explicitly closed.

**Comment by santanusetu** [comment]
Hi, I’d like to work on this.

I was able to reproduce the issue on master (`fb166d8d7`) and added an integration test. `OS_OPEN_STREAM` is reported when the `PrintStream` is constructed, even though the underlying `FileOutputStream` has already been closed. At the same time, the equivalent case where nothing is closed is still reported as expected.

While digging into it, I noticed the same false positive when a `FileOutputStream` is wrapped in a `BufferedOutputStream` or `PrintWriter`. That makes me think the underlying issue is how closing a wrapped stream is being tracked, so it may apply to stream wrappers more generally.

I’m happy to fix this at the wrapper level, but if you’d prefer to keep the scope limited to `PrintStream`, I can do that as well.
